src/main/java/pl/edu/mimuw/chess/Piece.java

```diff
@@ -0,0 +1,72 @@
+package pl.edu.mimuw.chess;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public abstract class Piece {
+
+  protected final Board board;
+  protected final Player owner;
+  protected XY position;
+
+  public Piece(Board board, Player owner, XY position) {
+    this.board = board;
+    this.owner = owner;
+    this.position = position;
+
+    board.putPiece(this);
+  }
+
+  public XY getPosition() {
+    return position;
+  }
+
+  public Player getOwner() {
+    return owner;
+  }
+
+  // Checks if the move would result in the piece leaving the board, or landing
+  // on top of a piece that cannot be captured.
+  protected boolean canLand(XY move) {
+    if (!((new XY(0, 0)).le(move) && move.lt(board.getDimensions())))
+      return false;
+    if (board.getPieceAt(move) == null) return true;
+    return board.getPieceAt(move).getOwner() != owner;
+  }
+
+  // All the moves the piece can make right now.
+  public Set<Move> getMoves() {
+    return new HashSet<>();
+  }
+
+  public void doMove(XY newPosition) throws GameEnded {
+    assert canLand(newPosition);
+    if (board.getPieceAt(newPosition) != null) {
+      assert board.getPieceAt(newPosition).getOwner() != owner;
+      board.getPieceAt(newPosition).beCaptured(this);
+    }
+    board.removePiece(this);
+    position = newPosition;
+    board.putPiece(this);
+  }
+
+  public void beCaptured(Piece byWhom) throws GameEnded {
+    assert owner != byWhom.owner;
+    board.removePiece(this);
+  }
+
+  public String getRepresentationIfWhite() {
```

Comment:
niepotrzebne, to klasa abtrakcyjna, wiec nie musi implementaować całego interfejsu i ta "domyślna" implementacja nie mauzasadnienia, a jedynie niewłaściwie zwalnie konkretne bierki z posiadania implementacji